transport.go

```diff
@@ -62,4 +69,6 @@ type Transport interface {
 	// Shutdown is called when memberlist is shutting down; this gives the
 	// transport a chance to clean up any listeners.
 	Shutdown() error
+
+	ProtocolType() ProtocolType
```

Comment:
Note this is a breaking change to the exported interface. Any custom implementations of the interface will no longer implement this interface.

An alternative would be making a new interface like:
```
type TypedTransport interface {
  Transport
  ProtocolType() ProtocolType
}
```
Then you just have to do optional type assertions to see if the transport implements the extra method. I recently did this in memberlist when i had to change `Broadcast`: https://github.com/hashicorp/memberlist/blob/master/queue.go#L128